### ISSUE TITLE
update to use conditional invoke

### ISSUE THREAD

**Issue by DickBaker** [dotnet/docs#15758]
in code example
if (SampleEvent != null)
            SampleEvent(this, new SampleEventArgs("Hello"));
simplify to
            SampleEvent?.Invoke(this, new SampleEventArgs("Hello"));
but also remarks should mention multi-task considerations

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: f782a9c5-2dc8-d8d1-b17d-246ab8524863
* Version Independent ID: f6f825c5-c64b-d13b-1c7e-a01bfb93093d
* Content: [event - C# Reference](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/event?f1url=https%3A%2F%2Fmsdn.microsoft.com%2Fquery%2Fdev16.query%3FappId%3DDev16IDEF1%26l%3DEN-US%26k%3Dk(event_CSharpKeyword)%3Bk(TargetFrameworkMoniker-.NETFramework%2CVersion%3Dv4.5.2)%3Bk(DevLang-csharp)%26rd%3Dtrue%26f%3D255%26MSPPError%3D-2147217396)
* Content Source: [docs/csharp/language-reference/keywords/event.md](https://github.com/dotnet/docs/blob/master/docs/csharp/language-reference/keywords/event.md)
* Product: **dotnet-csharp**
* Technology: **csharp-language-reference**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

**Comment by Thraka** [comment]
Thank you for opening this issue. I'll add it to the backlog. If you would like to contribute, you can submit a fix by editing the code. The code is contained in this file:

https://github.com/dotnet/samples/blob/master/snippets/csharp/VS_Snippets_VBCSharp/csrefKeywordsModifiers/CS/csrefKeywordsModifiers.cs

Thanks again!

**Comment by matthew45052** [comment]
This example above was confusing when I first saw it, because ?. is not a normal C/C++ operator, and it's not even an operator in older versions of C# either.

It wasn't until I got to the ?. operator documentation (https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/member-access-operators) that it made sense.  If you're going to teach events, you should first teach the simple case:

var handler = this.PropertyChanged;
if (handler != null)
{
    handler(…);
}

Worry about the fancy new ?. operator later.

**Comment by Thraka** [comment]
Hi @matthew45052 Thank you for the comment. We've opted to use the best built-in operator for the case. Considering the operator was added to the language 5 years ago, was created to handle the situation you're describing, it's the best choice.

However, you do bring up something I think that should be addressed. Regardless of the operator or the explicit null check, it should be noted in the article that the event is null until it's subscribed to, and that a null value should be checked for.

I'm creating an issue to track that. I'm unsure why this issue wasn't closed. Thank you for your feedback!
